poly/src/domain/radix2/fft.rs

```diff
@@ -138,20 +138,29 @@ impl<F: FftField> Radix2EvaluationDomain<F> {
     }
 
     fn io_helper<T: DomainCoeff<F>>(&self, xi: &mut [T], root: F) {
-        let roots = self.roots_of_unity(root);
+        // Take the roots of unity, and we will adjust this vector
+        // as we do rounds of the fft, in order to keep the roots cache aligned.
+        let mut cache_aligned_roots = self.roots_of_unity(root);
+        let mut root_stride = 1;
 
         let mut gap = xi.len() / 2;
         while gap > 0 {
             // each butterfly cluster uses 2*gap positions
             let chunk_size = 2 * gap;
-            let nchunks = xi.len() / chunk_size;
+            let root_len = cache_aligned_roots.len();
+            if root_stride > MAX_ROOT_STRIDE && root_len > ROOT_STOP_RESIZING {
+                for i in 0..(root_len / root_stride) {
+                    cache_aligned_roots[i] = cache_aligned_roots[i * root_stride];
```

Comment:
Is there a way to do a loop like this with rayon, without additional memory allocations?

Reply:
Maybe not, since it can't really be split into segments nicely

Reply:
To be in place, mustn't this be sequential? For example, because we must read `cache_aligned_roots[1 * 4]` before we overwrite `cache_aligned_roots[4]`

(also, the first iteration just copies 0 into itself, so maybe we want to start at 1 instead of 0)